src/_pytest/assertion/util.py

```diff
@@ -436,16 +478,20 @@ def _compare_eq_cls(
     if same and verbose < 2:
         explanation.append("Omitting %s identical items, use -vv to show" % len(same))
     elif same:
-        explanation += ["Matching attributes:"]
-        explanation += pprint.pformat(same).splitlines()
+        explanation += ["{}Matching attributes:".format(indent)]
+        mystr = indent + str(pprint.pformat(same).splitlines())
```

Comment:
Here, instead of calling `str` on the list, we should instead indent every line on its own, and append them to `explanation`, like before. Otherwise the attributes are shown with some repr format (this is why you had to change a bunch of unrelated parts ot the tests).